sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -448,7 +448,11 @@ private bool ValidateTokenSignature(AttestationSigner[] possibleSigners)
             {
                 // The leaf certificate is defined as the certificate which signed the token, so we just need to look
                 // at the first certificate in the chain.
+#if NET6_0_OR_GREATER
+                AsymmetricAlgorithm asymmetricAlgorithm = (AsymmetricAlgorithm)signer.SigningCertificates[0].GetRSAPublicKey() ?? signer.SigningCertificates[0].GetDSAPublicKey();
+#else
                 AsymmetricAlgorithm asymmetricAlgorithm = signer.SigningCertificates[0].PublicKey.Key;
```

Comment:
See: [SYSLIB0027: PublicKey.Key is obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0027)